sdk-base/src/main/java/com/mapbox/maps/plugin/viewport/experimental/data/FollowingViewportDataSource.kt

```diff
@@ -0,0 +1,5 @@
+package com.mapbox.maps.plugin.viewport.experimental.data
+
+interface FollowingViewportDataSource : ViewportDataSource {
+  var options: ViewportDataSourceOptions
+}
```

Comment:
If `FollowingViewportDataSource` is our opinionated default type, we shouldn't need to open up this interface (or make it an interface in the first place). Same with `OverviewViewportDataSource`.

Reply:
We expose this interface to allow user customising the ViewportDataSourceOptions. And I have updated this PR after sync with @macdrevx , please review again.